### PR TITLE
Search: Fix checkbox and row styles

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

Some recent style changes had changed what search looked like, this tweaks those a little bit to revert them to the previous style.
Specifically it fixes the checkbox alignment and the size of the search row

**Why do we need this feature?**

This is what the search looked like before:
Screenshot 2023-03-29 at 12 01 30

After:
Screenshot 2023-03-29 at 12 00 29

**Who is this feature for?**

Anyone

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
- [ ] There are no known compatibility issues with older supported versions of Grafana, or plugins.
- [ ] It passes the [Hosted Grafana feature readiness review](https://docs.google.com/document/d/1QL9Ly8KnXzpb6ISbg49pTODRO5mhA5tkkfIZVX6pqQU/edit) for observability, scalability, performance, and security.

### CODE REVIEW

**Comment by joshhunt** [comment]
Haven't checked this out yet (i will today!), but the alignment also regressed in the folder view 
image

Does changing Checkbox to inline-grid also address that?

**Comment by JoaoSilvaGrafana** [comment]
> Haven't checked this out yet (i will today!), but the alignment also regressed in the folder view image
> 
> Does changing Checkbox to inline-grid also address that?

Yep it does!

Ops:
Screenshot 2023-03-29 at 12 41 16

Locally:
Screenshot 2023-03-29 at 12 41 20

File: public/app/features/search/page/components/SearchResultsTable.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const SearchResultsTable = React.memo(
   }: SearchResultsProps) => {
     const styles = useStyles2(getStyles);
     const columnStyles = useStyles2(getColumnStyles);
-    const tableStyles = useTableStyles(useTheme2(), TableCellHeight.Md);
```
[the remark this answers is not in the document]
Comment (JoaoSilvaGrafana) [reply public/app/features/search/page/components/SearchResultsTable.tsx:55]: What do you mean doesn't work properly? Do you mean the second argument of this hook?

**Review by joshhunt — APPROVED** [review]
lgtm well done!

File: public/app/features/search/page/components/SearchResultsTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tskarhed) [reply public/app/features/search/page/components/SearchResultsTable.tsx:55]: Sorry for being unclear. `TableCellHeight.Md` sets the proper height, but the alignment of the content is weird. Is this the expected behavior for TableStyles?

[the remark this answers is not in the document]
Comment (JoaoSilvaGrafana) [reply public/app/features/search/page/components/SearchResultsTable.tsx:55]: This second argument was added recently, I haven't really properly checked what effects it has, but the correct row height is the equivalent of `TableCellHeight.Sm`. Maybe they are letting each cell content to align itself so it is not too opinionated? 🤷  But either way I dont think that should be part of this fix.

**Pull request merged**